### DISCUSSION TITLE
Support config mult cdn path

### DISCUSSION THREAD

**Discussion by peng** [withastro/roadmap#818] [Proposal]
### Body

# Summary

I want to use different cdn url prefix with static file, like image, js, css..  But astro now not support.

# Background & Motivation

I want to config different cdn url prefix with static file in my project. But not support.

So my project image manual upload, css configured as `build.inlineStylesheets = 'always'`, only js files use the configured cdn address. 

We have discussed it in this issues: https://github.com/withastro/astro/issues/9700#issuecomment-1895463471

# Goals

A **concise, bulleted-list** outlining the intended goals of this RFC. 

- What are the exact problems that you are trying to solve with this RFC?

Support multiple cdn url prefix for static file.

- If this RFC isn't approved, these goals can be used to develop alternative solutions.

I only use my current config, image manual upload, css configured as `build.inlineStylesheets = 'always'`, only js files use the configured cdn address. 

# Example

If the proposal involves a new or changed API, then include a basic code example.
Otherwise, omit this section if it's not applicable.

Use `build. assetsPrefix ` option, current assetsPrefix is string type. Add function type.

```ts

// assetsPrefix function

import { defineConfig } from 'astro/config';

export default defineConfig({
  build: {
    assetsPrefix: (fileType) => {
      if (fileType === 'js') {
        return 'https://js.example.com/'
      } else if (fileType === 'css') {
        return 'https://css.example.com/'
      }

      return ''
    }
  }
});

```

`fileType` type is string,  value is file extension.  Function return is string type.

**Comment by yhnsw0711** [comment]
We also need this feature, and this configuration form is also good.

**Comment by bluwy** [comment]
Completed in https://github.com/withastro/astro/pull/10189
